kedro-airflow/kedro_airflow/grouping.py

```diff
@@ -21,7 +30,7 @@ def get_memory_datasets(catalog: DataCatalog, pipeline: Pipeline) -> set[str]:
 
 
 def create_adjacency_list(
-    catalog: DataCatalog, pipeline: Pipeline
+    catalog: CatalogProtocol[Any] | DataCatalog, pipeline: Pipeline
```

Comment:
qq - why do we need `CatalogProtocol[Any]` but not `CatalogProtocol`?

Reply:
The original `mypy` error said something like it's getting arguments of the type `CatalogProtocol[Any]` but seems like removing this works just fine with `CatalogProtocol` so i've reverted that